src-tauri/src/farmer.rs

```diff
@@ -59,43 +57,39 @@ pub(crate) async fn farming(path: String, reward_address: String, plot_size: u64
             dsn_sync: false,
         };
 
-        if let Err(err) = farm(path.clone().into(), farming_args.clone()).await {
-            error!("farm function failed to start, with error: {err}");
-        }
```

Comment:
Well, this should have returned an error back, so you can render it on the frontend instead of moving it completely into background that will spin forever in an infinite loop printing gigabytes of errors once error happens. Do I understand that now user will have no indication that farmer isn't working at all unless they look into logs?

Reply:
> Do I understand that now user will have no indication that farmer isn't working at all unless they look into logs?

it is actually the same case as before, nothing changed about that. 
1. Frontend is invoking the `startFarming` function, and backend gives back a result regarding the farmer started successfully or not. 
2. then, somehow if farming crashes (due to node restart for example), we are trying to restart the farmer. If these will fail continuously, yes, frontend won't have a direct way of knowing it. 

Till now, frontend only knew about the first start of the farmer succeeded or not (and it is still like that).

Reply:
Yes, but you could have replaced `if let Err` with `?` and it would return an error, now everything happens in the background instead, so no way to catch start failure.

Reply:
I see your point. 
I will work on it. In the meantime, we will still not be able to tell frontend if the subsequent farmer instances fail. If we want to solve this one too, maybe another approach will be required.

Reply:
how about this?
https://github.com/subspace/subspace-desktop/blob/farmer-simplify-suggestions/src-tauri/src/farmer.rs#L60-L88

Reply:
Yes, just add some message to `error!("{error}")` so it is easier to find context.